### DISCUSSION TITLE
brew install vimac throws me Error: vimac: no bottle available!

### DISCUSSION THREAD

**Discussion by rustiever** [Homebrew/discussions#987] [Casks]
Sorry if this is dumb question, 
I followed the documentation and I got the error as mentioned above
and here is my cask file
```ruby
class Vimac < Formula
  version "0.3.15"
  desc "productivity application that revolutionizes the way you navigate macOS"
  homepage "https://vimacapp.com"
  # hostname is different
  url "https://appcenter-filemanagement-distrib4ede6f06e.azureedge.net/43c02f6d-bf67-47c4-8845-6fcb92ace0c3/Vimac_distribution.zip?sv=2019-02-02&sr=c&sig=rt8b1IJuDZmhfpITTr5liL3OesQUmEUQqcwtbih%2Fsy0%3D&se=2021-03-05T12%3A02%3A11Z&sp=r"
  sha256 "8c206518d8cc112cd771dac93ff8079a0065dc3315c039a2fafc5a81d5883aa6"
  license "GPL-3.0 License"

  #app "Vimac.app"

  def install
    # Remove unrecognized options if warned by configure
    system "./configure", "--disable-debug",
                          "--disable-dependency-tracking",
                          "--disable-silent-rules",
                          "--prefix=#{prefix}"
    # system "cmake", ".", *std_cmake_args
  end

  test do
    # `test do` will create, run in and delete a temporary directory.
    #
    # This test will fail and we won't accept that! For Homebrew/homebrew-core
    # this will need to be a test that verifies the functionality of the
    # software. Run the test with `brew test vimac`. Options passed
    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
    #
    # The installed folder is not in the path, so use the entire path to any
    # executables being tested: `system "#{bin}/program", "do", "something"`.
    system "false"
  end
end
```
please tell me what I'm doing wrong

**Comment by SMillerDev** [comment]
```ruby
class Vimac < Formula
```
This is not a cask, it's a formula.

---

edit to clarify:

This is less mixed up then I initially thought it was. You're just mixing up terminology.

- cask: prebuilt binary on some website somewhere that brew generally just moves to the `/Applications` folder
- formula: a source-build piece of software that homebrew can make available as a binary (bottle).

Since you wrote a formula homebrew will try and install it as a formula. This means it will try and install the binary (bottle), unless you pass `--build-from-source` to the install command.

**Reply by rustiever** [reply]
> ```ruby
> ```ruby
> class Vimac < Formula
> ```
> 
> 
> This is not a cask, it's a formula.
> 
> edit to clarify:
> This is less mixed up then I initially thought it was. You're just mixing up terminology.
> 
> cask: prebuilt binary on some website somewhere that brew generally just moves to the /Applications folder
> formula: a source-build piece of software that homebrew can make available as a binary (bottle).
> 
> Since you wrote a formula homebrew will try and install it as a formula. This means it will try and install the binary (bottle), unless you pass --build-from-source to the install command.
> ```

ohh thanks for the response

**Comment by rustiever** [comment]
but ```brew create --cask download-url --set-name my-new-cask``` running this command I got above template
what you think about that?

**Reply by gromgit** [reply]
> but `brew create --cask download-url --set-name my-new-cask` running this command I got above template
> what you think about that?

Doing that created a _cask_ template on my box, so odds are good that something's wrong with your setup.

What's the output of `brew config` and `brew doctor`?

**Reply by rustiever** [reply]
actually everything works now
thank you

**Comment by rustiever** [comment]
now the problem for me 
```
brew audit --new-cask vimac
==> Downloading https://appcenter-filemanagement-distrib3ede6f06e.azureedge.net/05dc3913-5945-4c0e-b671-25e535801364/Vimac_distribution.zip?sv=2019-02-02&sr=c&s
Already downloaded: /Users/sharan/Library/Caches/Homebrew/downloads/4659d9bd7bec643f267e98bbeb59fb98767b547907508992aa8cee73264ba30e--Vimac_distribution.zip
Warning: No checksum defined for cask 'vimac', skipping verification.
audit for vimac: failed
 - cask token mentions platform
 - The URL https://vimacapp.com is not reachable
Error: 2 problems in 1 cask detected
```
my cask file
```ruby
cask "vimac" do
  version "0.3.15 (25)"
  sha256 :no_check
  url "https://appcenter-filemanagement-distrib3ede6f06e.azureedge.net/05dc3913-5945-4c0e-b671-25e535801364/Vimac_distribution.zip?sv=2019-02-02&sr=c&sig=70Ombv8RB62apFWMzBfr%2FuW%2BDzWJb4TZ2vtlTS2H3xw%3D&se=2021-03-06T07%3A57%3A39Z&sp=r&download_origin=appcenter",
      verified: "appcenter-filemanagement-distrib3ede6f06e.azureedge.net/05dc3913-5945-4c0e-b671-25e535801364/"
  name "Vimac"
  desc "productivity application that revolutionizes the way you navigate macOS"
  homepage "https://vimacapp.com"

  app "Vimac.app"
end
```

**Reply by rustiever** [reply]
please tell me what should I do
